bin/p2-preparer/main.go

```diff
@@ -85,6 +90,44 @@ func main() {
 		logger.WithError(err).Fatalf("Could not do initial build reality at launch: %s", err)
 	}
 
+	whiteListPods := make(map[types.PodID]bool)
+	// Keep loopinng the white list of pods in pod whitelist file, and install the non existing pods
+	// exit while the white list is empty
+	if preparerConfig.PodWhitelistFile != "" {
+		for {
+			_, err := os.Stat(preparerConfig.PodWhitelistFile)
+			if os.IsNotExist(err) {
+				logger.WithError(err).Warningf("Pod whilelist path is not valid")
+			}
+			if err != nil {
+				logger.WithError(err).Warningf("Could not check for pod whitelist file's existence")
+			}
+
+			podWhitelist, err := loadTokens(preparerConfig.PodWhitelistFile)
+			if err != nil {
+				logger.WithError(err).WithField("path", preparerConfig.PodWhitelistFile).Fatalln("Could not read pod whitelist file")
+			}
+
+			// if the pod white list is empty, jump out of the loop, then p2-prepare proceeds to start with full functionality
+			if len(podWhitelist) == 0 {
+				break
+			}
+
+			for pod := range podWhitelist {
+				whiteListPods[types.PodID(pod)] = true
+			}
+
+			if len(whiteListPods) != 0 {
```

Comment:
What is the purpose of this check? Is it possible for the number of white listed pods to become 0 between the check on line 112 and here?

Reply:
good catch!